irc.go

```diff
@@ -54,6 +54,89 @@ func (ms *modeSet) Apply(s string) error {
 	return nil
 }
 
+type channelModeType byte
+
+const (
+	modeTypeA channelModeType = iota
+	modeTypeB
+	modeTypeC
+	modeTypeD
+)
+
+var stdChannelModes = map[byte]channelModeType{
+	'b': modeTypeA,
+	'e': modeTypeA,
+	'I': modeTypeA,
+	'k': modeTypeB,
+	'l': modeTypeC,
+	'i': modeTypeD,
+	'm': modeTypeD,
+	'n': modeTypeD,
+	's': modeTypeD,
+	't': modeTypeD,
+}
+
+func ApplyChannelModes(modeTypes map[byte]channelModeType, modes map[byte]string, modeStr string, arguments ...string) error {
+	nextArgument := 0
```

Comment:
We should use a consistent naming: either argument or param. Here we have `arguments`, `param`, `nextArgument`.

Reply:
Right. I've picked argument, for "mode argument" (as written in the specs)